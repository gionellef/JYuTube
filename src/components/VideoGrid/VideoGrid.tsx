import * as React from 'react';

import { Divider } from 'semantic-ui-react';
import { VideoPreview } from '../VideoPreview/VideoPreview';
import VideoGridHeader from './VideoGridHeader/VideoGridHeader';

import './VideoGrid.scss';

interface VideoGridProps {
  hideDivider?: boolean,
  title?: string,
};

const VideoGrid : React.FC<VideoGridProps> = (props) => {
  const divider = props.hideDivider ? null : <Divider />;

  return (
    <React.Fragment>
      <VideoGridHeader title={ props.title } />
      <div className='video-grid'>
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
      </div>
      { divider }
    </React.Fragment>
  );
}

export default VideoGrid;