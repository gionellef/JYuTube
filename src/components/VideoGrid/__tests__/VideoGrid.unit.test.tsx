import * as React from 'react';
import { shallow } from 'enzyme';

import VideoGrid from '../VideoGrid';

describe('VideoGrid', () => {

  test('renders VideoGrid without props', () => {
    const wrapper = shallow(<VideoGrid />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders VideoGrid with hideDivider prop false', () => {
    const wrapper = shallow(<VideoGrid hideDivider={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders VideoGrid with title', () => {
    const wrapper = shallow(<VideoGrid title='Trending' />);
    expect(wrapper).toMatchSnapshot();
  });
});