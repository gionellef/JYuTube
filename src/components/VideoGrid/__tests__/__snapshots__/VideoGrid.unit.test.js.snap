// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders VideoGrid with divider 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
</Fragment>
`;

exports[`VideoGrid renders VideoGrid with divider title 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
</Fragment>
`;

exports[`VideoGrid renders VideoGrid with hideDivider 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders VideoGrid with hideDivider prop false 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders VideoGrid with hideDivider prope 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders VideoGrid with title 1`] = `
<Fragment>
  <VideoGridHeader
    title="Trending"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders VideoGrid without props 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;
